### PR TITLE
elm: fix head and spin off elm-reactor

### PR DESCRIPTION
**Pull request by ilovezfs**

### CODE REVIEW

**Comment by ilovezfs** [comment]
@mikemcquaid This is just proof of concept that now all components can be compiled from their respective HEADay successfully. I can reintegrate elm-reactor into the original formula (either as an option or not) if you'd like, or have it be a separate formula. Note that as it is here, it doesn't depend_on elm so the user would want to install that as well, or an elm compiler from elsewhere, or I could install the vendored tools into elm_reactor's libexec, or something else. Any preferences?

File: Library/Formula/elm-reactor.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  homepage "http://elm-lang.org"
+  url "https://github.com/elm-lang/elm-reactor/archive/0.16.tar.gz"
+  sha256 "fd105180b92364c3ba8df58bfa88306adb8a21074786267ab1eaa634777ded7f"
+  head "https://github.com/elm-lang/elm-reactor.git"
+
+  bottle do
+    sha256 "af4973a6501741e69289163297cb1f3f20f1837dbca7c38ca8fc7cf8116a8460" => :el_capitan
+    sha256 "8e768f845671c3e83197abd99cb4e07bea0c2cdaa293023fe27fbb7b5f90cb19" => :yosemite
+    sha256 "631587361e3e35dbd6677de2c6b580bd88b5ed04835c8b6ca763649d98579da7" => :mavericks
+  end
```
Comment (mikemcquaid) [inline Library/Formula/elm-reactor.rb]: Where do these come from?

File: Library/Formula/elm-reactor.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      resource(r).stage buildpath/r
+    end
+
+    cabal_sandbox do
+      cabal_sandbox_add_source "elm-reactor", *resources
+      cabal_install "--only-dependencies", *resources, "elm-reactor"
+      cabal_install *resources
+      ENV.deparallelize { cabal_install "--prefix=#{prefix}", "elm-reactor" }
+    end
+  end
```
Comment (mikemcquaid) [inline Library/Formula/elm-reactor.rb]: Can a test be added to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Comment by mikemcquaid** [comment]
Feels like it should depend on elm. What's the desire to separate them, out of interest?

File: Library/Formula/elm-reactor.rb (hunk 1 above)
Comment (ilovezfs) [inline Library/Formula/elm-reactor.rb]: Those are just the bottles from elm. I wasn't sure whether to leave in a bottle block. I guess new formulae are supposed to have no bottle block (or an empty bottle block?) until the bottle bot kicks in?

Comment (mikemcquaid) [inline Library/Formula/elm-reactor.rb]: > I guess new formulae are supposed to have no bottle block (or an empty bottle block?) until the bottle bot kicks in?

Yup :+1:

**Comment by ilovezfs** [comment]
elm-compiler (dev) is at 0.17: https://github.com/elm-lang/elm-compiler/commit/8d8afdacdd9437e2fa46a5c9e8e425e97409aec5

elm-package (master) is at 0.17: https://github.com/elm-lang/elm-package/commit/736b7540aa14cb8cfdb2738e62c471d4682c3263 and needs elm-compiler 0.17

elm-make (master) is at 0.17: https://github.com/elm-lang/elm-make/commit/d572c04fc9b5ee1d2c091be421fea04cb296e34a and needs elm-compiler 0.17


Whereas elm-reactor is at 0.16 and requires the 0.16 versions of elm-compiler, elm-package, and elm-make.

So the only thing standing in the way of a successful elm 0.17 HEAD build (HEAD because there aren't 0.17 tags yet) is the elm.rb's theory that they all belong together in a single formula, but that doesn't actually seem to be the upstream development pattern, since the latest development of elm-reactor in the `next` and `expando` branches still relies on elm compiler 0.16.

So elm-reactor could depend on elm, but it wouldn't be able to build without the vendored elm-compiler, elm-make, elm-package, unless elm were a non-HEAD build, which pretty much defeats the purpose of separating them.

They could all be combined back into a single formula, assuming that the 0.16 versions were vendored in for both stable and HEAD builds, just for the sake of building elm-reactor, while also building the 0.17 versions of everything-except-reactor.

The simplest answer of course is to just remove the elm head altogether and wait for them to figure out how to get elm-reactor to compile with the 0.17 tool chain, and to create 0.17 tags of everything, but it seemed somewhat disappointing to junk a perfectly good elm 0.17 head build just because of upstream's neglect of the elm-reactor component, and the brew formula's assumption that all of the components are actually being developed in parallel, when that's clearly not the case.

File: Library/Formula/elm-reactor.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  sha256 "fd105180b92364c3ba8df58bfa88306adb8a21074786267ab1eaa634777ded7f"
+  head "https://github.com/elm-lang/elm-reactor.git"
+
+  bottle do
+    sha256 "af4973a6501741e69289163297cb1f3f20f1837dbca7c38ca8fc7cf8116a8460" => :el_capitan
+    sha256 "8e768f845671c3e83197abd99cb4e07bea0c2cdaa293023fe27fbb7b5f90cb19" => :yosemite
+    sha256 "631587361e3e35dbd6677de2c6b580bd88b5ed04835c8b6ca763649d98579da7" => :mavericks
+  end
+
+  depends_on "ghc" => :build
```
Comment (ilovezfs) [inline Library/Formula/elm-reactor.rb]: Should this be removed since cabal-install depends on ghc anyway?

**Comment by ilovezfs** [comment]
Strange. Trying to run the regular test for elm, I get 
```
Error: failed with 'FailedConnectionException2 "package.elm-lang.org" 1234 False connect: does not exist (Connection refused)' when sending request to
    <http://package.elm-lang.org:1234/all-packages?elm-package-version=0.17>
```
so I'm closing the PR since it seems they don't intend anyone to use head yet. Unless someone knows a workaround, I think we should probably just remove the head build.

**Pull request closed without merging**